### ISSUE TITLE
clear_config_item leaves network key empty on config file

### ISSUE THREAD

**Issue by tripledes** [lxc/lxc#559]
I'm able to reproduce the issue from both Python and Ruby, with the following steps:
- Container configuration:

```
lxc.include = /usr/share/lxc/config/ubuntu.common.conf
# Container specific configuration
lxc.rootfs = /var/lib/lxc/ubuntu_test/rootfs
lxc.utsname = ubuntu_test
lxc.arch = amd64
# Network configuration
lxc.network.type = veth
lxc.network.link = lxcbr0
lxc.network.flags = up
lxc.network.hwaddr = 00:16:3e:9f:26:7c
lxc.network.0.ipv4 = 10.0.3.3/24
```
- Reproduction from python3 interpreter:

```
Python 3.2.3 (default, Feb 27 2014, 21:31:18) 
[GCC 4.6.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import lxc
>>> c = lxc.Container('ubuntu_test')
>>> c.state
'RUNNING'
>>> c.get_config_item('lxc.network.0.ipv4')
['10.0.3.3/24']
>>> c.clear_config_item('lxc.network.0.ipv4')
True
>>> c.set_config_item('lxc.network.0.ipv4','10.0.3.4/24')
True
>>> c.save_config()
True
```
- Configuration file after applying the changes from python interpreter:

```
lxc.include = /usr/share/lxc/config/ubuntu.common.conf
# Container specific configuration
lxc.rootfs = /var/lib/lxc/ubuntu_test/rootfs
lxc.utsname = ubuntu_test
lxc.arch = amd64
# Network configuration
lxc.network.type = veth
lxc.network.link = lxcbr0
lxc.network.flags = up
lxc.network.hwaddr = 00:16:3e:9f:26:7c
lxc.network.0.ipv4 = 
lxc.network.0.ipv4 = 10.0.3.4/24
```

Having an empty key, produces parsing errors when trying to make more modifications from both Python and Ruby bindings.

This issue has been tested on Ubuntu Precise using the official PPA with latest liblxc 1.1.2 version.

I'm sorry to say I'm a bit short on time to try to produce a patch, I'd be very grateful if anyone could have a look as it's currently breaking my puppet lxc module, see tripledes/sjimenez-lxc#4.

Thanks!

**Comment by stgraber** [comment]
Oh, interesting, so the generated config above, while ugly should be parsed as valid. It's unfortunately not the first bug we've found in that part of the parser...

**Comment by tripledes** [comment]
I see, I was focusing on the clear_config_item() part...The following is the error I get from Ruby bindings:
```
lxc: confile.c: config_network_ipv4: 752 File exists - invalid ipv4 address: 
lxc: parse.c: lxc_file_for_each_line: 57 Failed to parse config: lxc.network.0.ipv4 = 
```

**Comment by tripledes** [comment]
The issue kind of makes sense, I mean, the error comes from https://github.com/lxc/lxc/blob/master/src/lxc/confile.c#L759-765, it's a validation issue which IMHO makes sense.

Do you still think the configuration should be parsed as valid? If so, I guess the way should be to check whether the value at the right side of the equal symbol is empty, if so, just skip validation. On the other hand I still believe the clear_config_item shouldn't leave the value empty :-)

Thoughts?

**Comment by stgraber** [comment]
= with nothing after it should be parsed as "reset whatever value is stored in there". That's how it's parsed for just about everything else.

**Comment by irnes** [comment]
This bug is still present in v2.0.1

Removing a network interface just add the following line:
**lxc.network.0 =**
at the bottom of the network interface configuration and produce the following error:
  
lxc-ls: parse.c: lxc_file_for_each_line: 57 Failed to parse config: lxc.network.0 =

**Comment by brauner** [comment]
This is fixed by 0797e123e0b489fed5fadcb4ab06ab5277468b85 .
